### PR TITLE
fix(apply-project): Resource role name in DTO is modified but be trusted

### LINKED ISSUE #759: [Bug]: Resource role name in DTO should not be trusted
### ODC version

4.2.3

### OB version

null

### What happened?

Resource role name in DTO is trusted by ODC

### What did you expect to happen?

Resource role name in DTO should not be trusted

### How can we reproduce it (as minimally and precisely as possible)?

Start a apply project task;
Catch the request and modify the resource role name field, and then submit this request.

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-bug
module-apply project

#### Which issue(s) this PR fixes:
Resource role name in DTO could by modified but trusted by ODC. This PR fix it.
Fixes #759

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/permissionapply/project/ApplyProjectPreprocessor.java [hunk 1]
```diff
@@ -66,6 +66,10 @@ public void process(CreateFlowInstanceReq req) {
         List<ResourceRoleEntity> resourceRoleEntities = checkResourceRoleExist(
                 parameter.getResourceRoles().stream().map(ApplyResourceRole::getId).collect(Collectors.toList()));
         parameter.setUserId(authenticationFacade.currentUserId());
+        parameter.getProject().setName(projectEntity.getName());
+        for (int index = 0; index < parameter.getResourceRoles().size(); index++) {
+            parameter.getResourceRoles().get(index).setName(resourceRoleEntities.get(index).getRoleName());
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/permissionapply/project/ApplyProjectPreprocessor.java:71]: Why do you need the project role name?

Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/permissionapply/project/ApplyProjectPreprocessor.java:71]: The front-end should display it in task details. If we return it, the front-end will no need extra query.

**Review by MarkPotato777 — DISMISSED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**